File: ProjetAngular/src/app/Models/EmploisDuTemps.ts
import {Time} from "@angular/common";
import {Enseignants} from "./Enseignants";
import {Cours} from "./Cours";
import {Classes} from "./Classes";

export class EmploisDuTemps{
  constructor(
    public id: Number,
    public enseignant: Enseignants,
    public cours: Cours,
    public heureDebut: String,
    public heureFin: String,
    public classe: Classes
  ) {}


}

File: Projetemplois/src/main/java/com/emplois/temps/models/EmploisDuTemps.java
package com.emplois.temps.models;


import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalTime;

@Entity
@AllArgsConstructor
@Data
@NoArgsConstructor
public class EmploisDuTemps {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY )
    @Column(name="id")
    private Long id;

    private String heuredebut;

    private String heuredefin;

    @ManyToOne
    private Enseignants enseignants;

    @ManyToOne
    private Cours cours;

    @ManyToOne
    private Classes classes;

}
